lib/Zonemaster/CLI.pm

```diff
@@ -839,16 +847,46 @@ sub add_fake_delegation {
 
         if ( $ip ) {
             my $net_ip = Net::IP::XS->new( $ip );
-            if ( ( $ip =~ /($IPV4_RE)/ && Net::IP::XS::ip_is_ipv4( $ip ) )
+            unless (
+                ( $ip =~ /($IPV4_RE)/ && Net::IP::XS::ip_is_ipv4( $ip ) )
                 or
-                 ( $ip =~ /($IPV6_RE)/ && Net::IP::XS::ip_is_ipv6( $ip ) )
-            ) {
-                push @{ $data{ $name } }, $ip;
-            }
-            else {
+                ( $ip =~ /($IPV6_RE)/ && Net::IP::XS::ip_is_ipv6( $ip ) )
+            )
+	    {
                 die Net::IP::XS::Error() ? "Invalid IP address in --ns argument:\n\t". Net::IP::XS::Error() ."\n" : "Invalid IP address in --ns argument.\n";
             }
         }
+    }
+
+    return;
+}
+
+sub check_fake_ds {
+    my ( $self ) = @_;
+
+    foreach my $str ( @{ $self->ds } ) {
+        unless ( $str =~ /$DS_RE/ ) {
+            say STDERR __( "--ds ds data must be in the form \"keytag,algorithm,type,digest\"");
```

Comment:
I suggest
```
say STDERR __( "--ds ds data must be in the form \"keytag,algorithm,type,digest\". E.g. space is not permitted anywhere in the string.");
```
Motivation:

When `dig` lists a DS record is seems to always split the digest in two with a space character. I think that bind accepts the space character. E.g. (see right end):

```
;; ANSWER SECTION:
malmo.se.		3600	IN	DS	24604 8 2 2CDEBE0EAAA9906D8C2025D348C0A984825A2B58F5C67CCB7D6A5429 2C697107
malmo.se.		3600	IN	DS	24604 8 4 4249EE3A52E94F7B42C593D6ABD506A1413B2F06522F0455E58E99B6 263330A224F89576AE56B420603A9BE22EEC42FB
```